index.js

```diff
@@ -37,33 +63,58 @@ const getLuminance = (RGBarray) => {
   return luminance;
 };
 
-const twoHexesRatio = (color1, color2) => {
-  const RGBColor1 = hexToRGB(color1);
-  const RGBColor2 = hexToRGB(color2);
-  const color1Luminance = getLuminance(RGBColor1);
-  const color2Luminance = getLuminance(RGBColor2);
+const calculateRatio = (color1, color2) => {
+  const color1Luminance = getLuminance(color1);
+  const color2Luminance = getLuminance(color2);
   /* (L1 + 0.05) / (L2 + 0.05), whereby:
     L1 is the relative luminance of the lighter of the colors, and
     L2 is the relative luminance of the darker of the colors. */
-  let [lighterLum, darkerLum] = [color1Luminance, color2Luminance];
-  if (darkerLum > lighterLum) {
-    [lighterLum, darkerLum] = [darkerLum, lighterLum];
-  }
+  let lighterLum = Math.max(color1Luminance, color2Luminance);
+  let darkerLum = Math.min(color1Luminance, color2Luminance);
   return ((lighterLum + 0.05) / (darkerLum + 0.05)).toFixed(2);
 };
 
+const twoHexesRatio = (color1, color2) => {
+  const RGBColor1 = hexToRGB(color1);
+  const RGBColor2 = hexToRGB(color2);
+  return calculateRatio(RGBColor1, RGBColor2);
+};
+
+const twoRgbsRatio = (color1, color2) => {
+  const RGBColor1 = rgbToRGB(color1);
+  const RGBColor2 = rgbToRGB(color2);
+  return calculateRatio(RGBColor1, RGBColor2);
+};
+
+const twoRgbasRatio = (color1, color2) => {
+  const RGBColor1 = rgbaToCloseRGB(color1);
+  const RGBColor2 = rgbaToCloseRGB(color2);
+  return calculateRatio(RGBColor1, RGBColor2);
+};
+
```

Comment:
Now that we have the basic functionality working I think we can cut down on this repetition and just create one function. 

Something like this:
```
const colorFormatRatio = (color1, color2, convertRatio) => {
  const RGBColor1 = convertRatio(color1);
  const RGBColor2 = convertRatio(color2);
  return calculateRatio(RGBColor1, RGBColor2);
};
```
Then when you call the function you can just pass in the hexToRGB, or rgbToRGB as the third argument. 
```
 // CASE two RGBs
  if (rgbRegex.test(firstColor) && rgbRegex.test(secondColor)) {
    foregroundSwatch.style.backgroundColor = firstColor;
    backgroundSwatch.style.backgroundColor = secondColor;
    ratioResult.innerHTML = colorFormatRatio(firstColor, secondColor, rgbToRGB);
  }
```

Reply:
There are other small refactor changes we can make later on especially once the HSL format is added in.
But for now, I think making this small refactor would be a good move 👍